samples/nrf9160/modem_shell/src/sock/sock.c

```diff
@@ -956,17 +957,96 @@ int sock_send_data(
 	return 0;
 }
 
+static void sock_handle_poll_events(int socket_id, int poll_revents)
+{
+	struct sock_info *socket_info;
+	int buffer_size;
+
+	socket_info = &(sockets[socket_id]);
+
+	if (poll_revents & POLLIN) {
+		/* Data received. If receive buffer doesn't exist, allocate it. */
+		if (sock_receive_buffer == NULL) {
+			sock_receive_buffer = k_calloc(SOCK_RECEIVE_BUFFER_SIZE + 1, 1);
+			if (sock_receive_buffer == NULL) {
+				mosh_error(
+					"Out of memory while reserving "
+					"receive buffer of size %d bytes",
+					SOCK_RECEIVE_BUFFER_SIZE);
+				/* Returning here for simplicity although we could handle
+				 * other events too but if we cannot allocate receive buffer,
+				 * sock tool just won't work and application has other issues too.
+				 */
+				return;
+			}
+		}
+
+		/* Store timestamp for receiving data */
+		if (socket_info->recv_start_throughput) {
+			socket_info->start_time_ms = k_uptime_get();
+			socket_info->recv_start_throughput = false;
+		}
+
+		buffer_size = recv(
+			socket_info->fd, sock_receive_buffer, SOCK_RECEIVE_BUFFER_SIZE, 0);
+
+		if (buffer_size > 0) {
```

Comment:
Here is the only real change. Earlier this used to be as follows:
```
				if (buffer_size <= 0) {
					continue;
				}
```
I had to refactor the code because compliance didn't allow one more indentation level so that's why this small technical change shows a big delta.

Reply:
Thanks